src/sqlancer/citus/oracle/CitusNoRECOracle.java

```diff
@@ -1,14 +1,49 @@
 package sqlancer.citus.oracle;
 
+import java.sql.SQLException;
+import java.util.regex.Pattern;
+
+import sqlancer.Reproducer;
 import sqlancer.citus.gen.CitusCommon;
+import sqlancer.common.oracle.NoRECOracle;
+import sqlancer.common.oracle.TestOracle;
+import sqlancer.common.query.ExpectedErrors;
 import sqlancer.postgres.PostgresGlobalState;
-import sqlancer.postgres.oracle.PostgresNoRECOracle;
+import sqlancer.postgres.PostgresSchema;
+import sqlancer.postgres.PostgresSchema.PostgresColumn;
+import sqlancer.postgres.PostgresSchema.PostgresTable;
+import sqlancer.postgres.ast.PostgresExpression;
+import sqlancer.postgres.ast.PostgresJoin;
+import sqlancer.postgres.ast.PostgresSelect;
+import sqlancer.postgres.gen.PostgresCommon;
+import sqlancer.postgres.gen.PostgresExpressionGenerator;
 
-public class CitusNoRECOracle extends PostgresNoRECOracle {
+public class CitusNoRECOracle implements TestOracle<PostgresGlobalState> {
+    private final NoRECOracle<PostgresSelect, PostgresJoin, PostgresExpression, PostgresSchema, PostgresTable, PostgresColumn, PostgresGlobalState> oracle;
 
     public CitusNoRECOracle(PostgresGlobalState globalState) {
-        super(globalState);
-        CitusCommon.addCitusErrors(errors);
+        PostgresExpressionGenerator gen = new PostgresExpressionGenerator(globalState);
+        ExpectedErrors errors = ExpectedErrors.newErrors()
+                .with(PostgresCommon.getCommonExpressionErrors().toArray(new String[0]))
```

Comment:
What about overloading the `with` method to accept a `Collection`? This way, we might save a bit of boilerplate code at various places (I remember seeing this pattern before).

Reply:
Other than that, the change looks good to me.

Reply:
Btw, feel free to click on "Request review" at the top right so I get a notification when I should review something.

Reply:
I wanted to change this but I couldn't overload `with` with both `Collection<String>` and `Collection<Pattern>` due to type erasure, which is a little annoying. 

The alternative is to have separate methods `with` and `withRegex`. Do you think this is better?

Reply:
Could you just specify a generic type parameter `<T>` for the method and then a type-unsafe assignment (might need to add a warning suppression to the method)?

Reply:
Anway, merging this one first, as we could do this refactoring as part of a follow-up PR.